tensorflow/core/grappler/optimizers/mkl_remapper_test.cc

```diff
@@ -73,27 +75,51 @@ class MklRemapperTest : public GrapplerTest {
     if (add_op == kAddNOp) {
       auto addn = ops::AddN(s.WithOpName(add_op),
                             std::initializer_list<Input>{input_addn, bias_add});
-      if (has_relu) {
-        auto relu = ops::Relu(s.WithOpName("relu"), addn);
-        ops::Identity(s.WithOpName("fetch"), relu);
+      auto activate = s.WithOpName("activation");
+      auto fetch = s.WithOpName("fetch");
+      if (activation == "Relu") {
+        ops::Identity(fetch, ops::Relu(activate, addn));
+      } else if (activation == "Relu6") {
+        ops::Identity(fetch, ops::Relu6(activate, addn));
+      } else if (activation == "Elu") {
+        ops::Identity(fetch, ops::Elu(activate, addn));
+      } else if (activation == "LeakyRelu") {
+        ops::Identity(fetch, ops::internal::LeakyRelu(activate, addn));
       } else {
-        ops::Identity(s.WithOpName("fetch"), addn);
+        DCHECK(activation == "None");
+        ops::Identity(fetch, addn);
       }
     } else if (add_op == kAddV2Op) {
       auto add = ops::AddV2(s.WithOpName(add_op), input_addn, bias_add);
-      if (has_relu) {
-        auto relu = ops::Relu(s.WithOpName("relu"), add);
-        ops::Identity(s.WithOpName("fetch"), relu);
+      auto activate = s.WithOpName("activation");
+      auto fetch = s.WithOpName("fetch");
+      if (activation == "Relu") {
+        ops::Identity(fetch, ops::Relu(activate, add));
+      } else if (activation == "Relu6") {
+        ops::Identity(fetch, ops::Relu6(activate, add));
+      } else if (activation == "Elu") {
+        ops::Identity(fetch, ops::Elu(activate, add));
+      } else if (activation == "LeakyRelu") {
+        ops::Identity(fetch, ops::internal::LeakyRelu(activate, add));
       } else {
-        ops::Identity(s.WithOpName("fetch"), add);
+        DCHECK(activation == "None");
+        ops::Identity(fetch, add);
       }
```

Comment:
The logic is the same in all three cases (`AddN, AddV2, Add`). Can we refactor this to a function (or a lambda function)?

Reply:
Thanks for the advice. I rewrite this part of codes.